pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -63,6 +63,8 @@ import (
 	c "k8s.io/kubernetes/pkg/controller"
 )
 
+var mu = sync.Mutex{}
+
```

Comment:
I'm wondering if it'd be better to add a mutex as a member of the `Scheme` struct and use that internally in `AddKnownTypeWithName` when needed 🤔 
I feel like that'd be cleaner than introducing a global lock and passing it to the constructor and initializing it as part of `init`. wdyt?

Reply:
> I'm wondering if it'd be better to add a mutex as a member of the `Scheme` struct and use that internally in `AddKnownTypeWithName` when needed

That approach sounds good

Reply:
> I'm wondering if it'd be better to add a mutex as a member of the Scheme struct and use that internally in AddKnownTypeWithName when needed

> That approach sounds good

Actually, I'm now remembering what past me was thinking here.  Because the scheme is used for multi-threaded reads, it would take a fair amount of work to wrap everything.  If the tests were updated instead to have a `newTestScheme` function that returned a unique scheme for each test (the tests probably didn't intended to mutate a global scheme anyway), that will fix the issue without the fanout.

Reply:
Thanks Madhav and David. 

Function `NewTestScheme` has added to return a unique scheme for each test. PTAL